src/napari/layers/shapes/shapes.py

```diff
@@ -2458,7 +2457,7 @@ def interaction_box(self, index: int | Iterable[int]) -> BoxArray | None:
             the box, and the last point is the location of the rotation handle
             that can be used to rotate the box
         """
-        if isinstance(index, list | np.ndarray | set):
+        if isinstance(index, Iterable):
```

Comment:
I just spotted that the iterable does not guarantee that the object is sizeable, which we used in the next 4 lines. So I suggest adding a check if the object is `Sized` and if not, then converting it to a list. 

```suggestion
        if isinstance(index, Iterable):
            if not isinstance(index, Sized):
                index = list(index)
```

Reply:
Agree. Looks similar to how it was earlier with the ist conversion, but more abstract - I like it.